R/ssm_to_proteinpaint.R

```diff
@@ -0,0 +1,141 @@
+#' @title MAF to ProteinPaint table
+#' 
+#' @description This function takes a MAF-like data frame and convert it to the right format 
+#' for visualization with ProteinPaint.
+#' 
+#' @details For visualization with ProteinPaint, the output data frame must be saved to a 
+#' file and uploaded to https://proteinpaint.stjude.org/. 
+#' 
+#' For a valid ProteinPaint table for visualization, the MAF table must contain the required 
+#' columns to be converted to the ProteinPaint format. They are: Hugo_Symbol, RefSeq, 
+#' Chromosome, Start_Position, HGVSp_Short, Variant_Classification. Optional columns used 
+#' from the MAF table are: Mutation_Status, t_alt_count, t_depth, n_alt_count, n_depth, 
+#' Tumor_Sample_Barcode, Reference_Allele. In the case of using functions `get_ssm_by_samples`, 
+#' `get_coding_ssm`, or `get_ssm_by_patients` to get your MAF table and required/desired 
+#' columns are missing, consider using parameter `basic_columns = FALSE` to ensure that your 
+#' MAF has all needed columns. Other columns are taken from the metadata table generated 
+#' internally in `ssm_to_proteinpaint`.
+#' 
+#' @param maf A data frame in MAF format.
+#' @param print_removed_rows Boolean value indicating whether only removed rows, *i.e.* 
+#' rows in `maf` that do not contain values for required columns, should be printed. 
+#' Used for checking purpose.
+#'
+#' @return A data frame.
+#' 
+#' @import tidyr
+#' @export
+#'
+#' @examples
+#' library(GAMBLR.data)
+#' 
+#' # get maf
+#' my_maf = get_coding_ssm(basic_columns = FALSE)
+#' 
+#' # convert maf to ProteinPaint format
+#' pp_df = ssm_to_proteinpaint(my_maf)
+#' 
+ssm_to_proteinpaint = function(maf,
```

Comment:
Minor comment, but maybe we should harmonize all the parameter names that excpects a data frame in MAF format. I think, `maf_data` is the most widely used. What do you think?

Reply:
Good point. Agreed.